backend/src/controllers/auth.controller.ts

```diff
@@ -115,3 +116,49 @@ export const logout = async (req: Request, res: Response) => {
     }
   );
 };
+
+const client = new OAuth2Client();  
+export const googleLogin = async (req: Request, res: Response) => {
+  try{
+    const ticket = await client.verifyIdToken({
+      idToken: req.body.credential,
+      audience: process.env.GOOGLE_OAUTH_CLIENT_ID,
+    });
+    const payload = ticket.getPayload();
+    const email = payload?.email;
+
+    if (!email) {
+      logger.error('Email not found in payload');
+      return res
+        .status(HttpStatusCode.BadRequest)
+        .json({ message: 'Email not found in payload' });
+    }
+
+    let user = await getUserByIdentifier(email);
+    if (!user) {
+      const newUser: User = {
+        userName: payload?.name || '',
+        email: email,
+        password: '',
+        profilePicture: payload?.picture,
+      };
+      user = await createUser(newUser);
+    }
+    
+    const tokens = await getUserTokens(user);
+    logger.info('User logged in successfully');
+    res.status(HttpStatusCode.Ok).json(tokens);
+
+  } catch(error: any) {
+    logger.error('Error logging in witn Google: ', error.message);
+    if (error.message === errorMessages.INVALID_CREDENTIALS) {
+      logger.error('User not found or password is incorrect');
+      return res
+        .status(HttpStatusCode.Unauthorized)
+        .json({ message: errorMessages.USER_NOT_FOUND });
+    }
+    res
+      .status(HttpStatusCode.InternalServerError)
+      .json({ message: error.message });
+  }
+};
```

Comment:
This looks good but the issue that the business logic shouldn't be in the controller but in the service itself. Move the business logic to auth service in order to have a proper separation of concerns.

Controller:
```suggestion

export const googleLogin = async (req: Request, res: Response): Promise<void> => {
  try {
    const tokens = await googleLoginService(req.body.credential);
    logger.info('User logged in successfully');
    res.status(HttpStatusCode.Ok).json(tokens);
  } catch (error: any) {
    logger.error('Error logging in with Google: ', error.message);
    if (error.message === errorMessages.INVALID_CREDENTIALS) {
      logger.error('User not found or password is incorrect');
      return res.status(HttpStatusCode.Unauthorized).json({ message: errorMessages.USER_NOT_FOUND });
    }
    res.status(HttpStatusCode.InternalServerError).json({ message: error.message });
  }
};
```

Reply:
Done. moved into `auth.service.ts`, in function `googleLoginService`